### PR TITLE
expose suggestions::InferCtxtExt for clippy

### PR DESCRIPTION
**Pull request by rkuhn**
This is very useful to do good async/await diagnostic reporting, for example for https://github.com/rust-lang/rust-clippy/pull/5423.

### CODE REVIEW

File: src/librustc_trait_selection/traits/error_reporting/suggestions.rs [hunk 1]
```diff
@@ -22,7 +22,8 @@ use std::fmt;
 use super::InferCtxtPrivExt;
 use crate::traits::query::evaluate_obligation::InferCtxtExt as _;
 
-crate trait InferCtxtExt<'tcx> {
+// public for clippy
```
Comment (varkor) [inline src/librustc_trait_selection/traits/error_reporting/suggestions.rs:25]:
```suggestion
// This trait is public to expose the diagnostics methods to clippy.
```

**Comment by varkor** [comment]
This looks fine to me, but I'll pass on to the reviewer of https://github.com/rust-lang/rust-clippy/pull/5423 to make sure this is the right change (e.g. `note_obligation_cause` is available on `InferCtxtPrivExt` rather than `InferCtxtExt`).

r? @eddyb

**Comment by eddyb** [comment]
LGTM with @varkor's nit fixed.

**Comment by rkuhn** [comment]
Thanks guys!

**Pull request merged**